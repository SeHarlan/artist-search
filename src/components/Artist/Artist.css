.Artist {
  h1 {
    text-align: center;
    text-decoration: underline;
    font-size: 2rem;
  }
  h2 {
    text-align: center;
  }
  img {
    display: block;
    margin: 0 auto;
    width: 20vw;
    border: 4px outset wheat;
  }
  ul {
    height: 75vh;
    margin: 1rem 10vw;
    padding: 1rem;
    list-style: none;
    text-align: center;
    border-radius: 1rem;
    box-shadow: inset 0 0 10px rgba(0, 0, 0, 0.3);
    overflow: scroll;
  }
  li {
    margin: 0 5rem 5rem 5rem;
    padding-bottom: 2rem;
    border-bottom: 2px dotted wheat;
  }
  nav {
    width: 30vw;
    margin: 1rem auto;
    display: flex;
    justify-content: space-evenly;
    button {
      height: 2rem;
      padding: 0 1rem;
      border: 2px outset wheat;
      background-color: whitesmoke;
      font-family: inherit;
      border-radius: 0.6rem;
      outline: none;
      cursor: pointer;
    }
  }
}