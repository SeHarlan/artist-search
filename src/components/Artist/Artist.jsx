import React from 'react';
import { useFetchReleases } from '../../services/hooks';
import withList from '../HOCs/WithList/WithList';
import ReleaseItem from '../Release/ReleaseItem';
import { useParams } from 'react-router-dom';
import style from './Artist.css';

const Artist = () => {
  const { artist, artistId } = useParams();

  const { releases, pageNum, dec, inc } = useFetchReleases(artistId, artist);
  const Releases = withList(ReleaseItem);
  return (
    <section className={style.Artist}>
      <h1>{artist}</h1>
      <nav>
        <button disabled={pageNum === 1} onClick={dec}>Previous Page</button>
        <button disabled={releases.length < 20} onClick={inc}>Next Page</button>
      </nav>
      {releases.length ? <Releases list={releases} /> : <h2>Sorry, No Releases Found</h2>}
    </section>
  );
};

export default Artist;
