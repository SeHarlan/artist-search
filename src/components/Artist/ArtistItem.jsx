import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const ArtistItem = ({ artist, artistId, score, disambiguation = 'N/A' }) => {
  return (
    <Link to={`/artists/${artist}/${artistId}`}>
      <h2>{artist}</h2>
      <p>Match Score: {score}</p>
      <p>Disambiguation: {disambiguation}</p>
    </Link>
  );
};

ArtistItem.propTypes = {
  artist: PropTypes.string.isRequired,
  artistId: PropTypes.string.isRequired,
  score: PropTypes.number.isRequired,
  disambiguation: PropTypes.string
};

export default ArtistItem;
