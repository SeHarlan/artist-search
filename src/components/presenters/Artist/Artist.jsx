import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { useFetchReleases } from '../../../services/hooks';
import withList from '../../HOCs/WithList/WithList';
import ReleaseItem from '../Release/ReleaseItem';

const Artist = ({ match }) => {
  const [pageNum, setPageNum] = useState(1);

  const { artist, artistId } = match.params;

  const dec = () => setPageNum(prev => prev - 1);
  const inc = () => setPageNum(prev => prev + 1);

  const releasesData = useFetchReleases(artistId, pageNum, artist);
  const Releases = withList(ReleaseItem);
  return (
    <>
      <h1>{artist}</h1>
      <button disabled={pageNum === 1} onClick={dec}>Previous Page</button>
      <button disabled={releasesData.length < 20} onClick={inc}>Next Page</button>
      <Releases list={releasesData} />
    </>
  );
};

Artist.propTypes = {
  match: PropTypes.object.isRequired
};

export default Artist;
