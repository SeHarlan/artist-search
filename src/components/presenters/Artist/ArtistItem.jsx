import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const ArtistItem = ({ artist, artistId, score, country = 'N/A' }) => {
  return (
    <Link to={`/artists/${artist}/${artistId}`}>
      <h1>{artist}</h1>
      <p>Match Score: {score}</p>
      <p>country: {country}</p>
    </Link>
  );
};

ArtistItem.propTypes = {
  artist: PropTypes.string.isRequired,
  artistId: PropTypes.string.isRequired,
  score: PropTypes.number.isRequired,
  country: PropTypes.string
};

export default ArtistItem;
