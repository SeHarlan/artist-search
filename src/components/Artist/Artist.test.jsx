import React from 'react';
import { shallow } from 'enzyme';
import Artist from './Artist';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    artistId: 'test',
    artist: 'test',
  }),
  useRouteMatch: () => ({ url: '/artists/test/test' }),
}));

describe('Artist component', () => {
  it('renders Artist', () => {
    const wrapper = shallow(<Artist />);
    expect(wrapper).toMatchSnapshot();
  });
});
