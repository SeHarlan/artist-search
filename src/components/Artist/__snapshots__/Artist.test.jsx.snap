// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist component renders Artist 1`] = `
<section
  className="Artist"
>
  <h1>
    test
  </h1>
  <nav>
    <button
      disabled={true}
      onClick={[Function]}
    >
      Previous Page
    </button>
    <button
      disabled={true}
      onClick={[Function]}
    >
      Next Page
    </button>
  </nav>
  <h2>
    Sorry, No Releases Found
  </h2>
</section>
`;
